### ISSUE TITLE
[BUG]: Crash when trying to access instance field on the type itself

### ISSUE THREAD

**Issue by soraros** [modularml/mojo#1694]
### Bug description

As title. If the `let` is replaced with a `alias`, the parser doesn't crash and generates the correct error message.

### Steps to reproduce

Consider the following *invalid* code:

```mojo
struct S:
  var x: Int

fn main():
  let s = S
  print(S.x)
```

### System information

```shell
Mojo 0.7.0 on Docker, Intel Mac
```

**Comment by soraros** [comment]
Seems to be solved in 24.1.

**Comment by laszlokindrat** [comment]
Indeed, this is no longer an issue, since we disabled type variables while they are being redesigned. Please continue to open tickets if you run into problems!
